File: content/src/main/java/org/sandboxpowered/silica/minecraft/blocks/GlassPaneBlock.java
package org.sandboxpowered.silica.minecraft.blocks;

import org.sandboxpowered.api.block.BaseBlock;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.block.FluidLoggable;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.Properties;
import org.sandboxpowered.api.state.StateFactory;

public class GlassPaneBlock extends BaseBlock implements FluidLoggable {
    public GlassPaneBlock(Settings settings) {
        super(settings);
    }

    @Override
    protected BlockState createBaseState(BlockState baseState) {
        return baseState;
    }

    @Override
    public void appendProperties(StateFactory.Builder<Block, BlockState> builder) {
        super.appendProperties(builder);
        builder.add(Properties.NORTH);
        builder.add(Properties.EAST);
        builder.add(Properties.SOUTH);
        builder.add(Properties.WEST);
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/block/GlassPaneBlock.kt
package org.sandboxpowered.silica.block

import org.sandboxpowered.silica.state.StateProvider
import org.sandboxpowered.silica.state.block.BlockState
import org.sandboxpowered.silica.util.Identifier

class GlassPaneBlock(identifier: Identifier) : BaseBlock(identifier) {
    override fun appendProperties(builder: StateProvider.Builder<Block, BlockState>) {
        super.appendProperties(builder)
        builder.add(
            SilicaBlockProperties.EAST,
            SilicaBlockProperties.NORTH,
            SilicaBlockProperties.SOUTH,
            SilicaBlockProperties.WEST,
            SilicaBlockProperties.WATERLOGGED
        )
    }
}